simulation.py

```diff
@@ -0,0 +1,104 @@
+"""
+Fuctions for a process to simulate an alpha.
+"""
+
+import argparse
+import copy
+import importlib
+import os
+import pickle
+from thousandaire.constants import DATA_LIST_ALL
+from thousandaire.constants import OUTPUT_DIR, TRADING_INSTRUMENTS
+from thousandaire.constants import TRADING_CONFIGS, TRADING_REGIONS
+from thousandaire.data_loader import DataLoader
+from thousandaire.simulator import Simulator
+
+PRICE_DATASET = 'price_dataset'
+COST_FUNCTION = 'cost_function'
+
+def initialize():
+    """
+    Load all available dataset and bind them with workdays.
+
+    Return bound data, which is a dict of str (dataset names) -> Dataset.
+    """
+    raw_data = DataLoader(DATA_LIST_ALL).get_all()
+    workdays_all = {
+        region : raw_data['workdays'][region]
+        for region in TRADING_REGIONS if region in raw_data['workdays']}
+    bound_data = {}
+    for region, workdays in workdays_all.items():
+        bound_data[region] = copy.deepcopy(raw_data)
+        for dataset in bound_data[region].values():
+            dataset.set_workdays(workdays)
+    bound_data['workdays'] = workdays_all
+    return bound_data
+
+def extract_data(raw_data, data_list, price_dataset, region):
+    """
+    Extract data on data_list from raw_data.
+
+    Return data will be a dictionary, including three key-value pairs:
+    (1) 'workdays'
+        a DataController, the workdays data of the trading region.
+    (2) 'price'
+        a Dataset, the price dataset of the trading region.
+    (3) 'others'
+        a dict of str -> Dataset, other data that users specify in data_list.
+    """
+    return {
+        'workdays' : raw_data['workdays'][region],
+        'price_data' : raw_data[region][price_dataset],
+        'others' : {
+            name : raw_data[region][name]
+            for name in data_list if name in raw_data[region]}}
+
+def simulate(alpha_settings_path, skip_evaluation, output):
+    """
+    Act the process of simulating.
+    """
+    try:
+        settings = importlib.import_module(alpha_settings_path).AlphaSettings()
+    except FileNotFoundError as error:
+        raise FileNotFoundError("Alpha does not exist.") from error
+    except ImportError as error:
+        raise ImportError("No available AlphaSettings in %s"
+                          % alpha_settings_path) from error
+    data_all = initialize()
```

Comment:
Can you check the validity of alpha settings here?
We've implemented BaseAlphaSettings.is_valid to check whether all required fields are defined as expected types.
Besides, end_date is an optional field, and if it is not set by users, it implies that the simulation will run until the last available trading day. Such behavior must be implemented here.